net/keepalived/files/etc/keepalived/scripts/rsync.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/sh
+
+. /lib/functions.sh
+
+ha_sync_send() {
+	local cfg=$1
+	local address ssh_user ssh_port sync_list sync_dir sync_file
+
+	config_get address $cfg address
+	[ -z "$address" ] && return 0
+
+	config_get ssh_user $cfg ssh_user root
+	config_get ssh_port $cfg ssh_port 22
+	config_get ssh_key $cfg ssh_key /root/.ssh/id_rsa
+	config_get sync_dir $cfg sync_dir /usr/share/keepalived/rsync
+	config_get sync_list $cfg sync_list
+
+	for sync_file in $sync_list $(sysupgrade -l); do
+		[ -f "$sync_file" ] && {
+			dir="${sync_file%/*}"
+			list_contains FILES ${sync_file} || append FILES ${sync_file}
+		}
+		[ -d "$sync_file" ] && dir=${sync_file}
+		list_contains DIRS ${sync_dir}${dir} || append DIRS ${sync_dir}${dir}
+	done
+
+	changed_files=$(rsync --out-format='%n' --dry-run -a --relative $FILES -e "ssh -y -i $ssh_key -p $ssh_port" $ssh_user@$address:$sync_dir | wc -l)
+	[ $changed_files -le 0 ] && return 0
+
+	ssh -y -i $ssh_key -p $ssh_port $ssh_user@$address mkdir -p $DIRS
+	rsync -a --relative $FILES -e "ssh -y -i $ssh_key -p $ssh_port" $ssh_user@$address:$sync_dir
+
+	unset DIRS FILES
```

Comment:
Is that necessary? can't we make it locale variables?

Reply:
updated